### PR TITLE
Added UT for FormDataAnalytics controller.

### PR DESCRIPTION
**Pull request by yashgit891**
### - Link added should be a valid URL and accessible throughout the org.
   ### - If the branch name contains hotfix / revert by default the BVT workflow check will pass.

| Test Case Document URL                        |
|-----------------------------------------------|
| Please paste test case document link here.... |

### CODE REVIEW

File: Test/Unit/FormDataAnalyticsTest.php [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+        
+        $this->config->shouldReceive('getConfigData')->with('key_id')->andReturn('key_id');
+        $this->config->shouldReceive('getConfigData')->with('key_secret')->andReturn('key_secret');
+        
+        $this->trackPluginInstrumentation->shouldReceive('rzpTrackSegment')->andReturn(['status' => 'success']);
+        $this->trackPluginInstrumentation->shouldReceive('rzpTrackDataLake')->andReturn(['status' => 'success']);
+        
+        $this->resultFactoryMock->shouldReceive('create')->with('json')->andReturn($this->json);
+
+        $this->formDataAnalytics = \Mockery::mock(Razorpay\Magento\Controller\Payment\FormDataAnalytics::class, [   $this->context,
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php]: Remove space after bracket

File: Test/Unit/FormDataAnalyticsTest.php [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+            ]);
+        
+        $expectedResponse = '{"segment":{"status":"success"},"datalake":{"status":"success"}}';
+        $response = $this->formDataAnalytics->execute();
+        $this->assertSame($expectedResponse, $this->getProperty($response, 'json'));
+    }
+    public function testExecuteEventException()
+    {
+        $this->formDataAnalytics->shouldReceive('getPostData')->andReturn([
+            'properties'    => $this->eventProperties
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php]: Keep only one space after key

File: Test/Unit/FormDataAnalyticsTest.php [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+            ]);
+
+        $this->formDataAnalytics->execute();
+    }
+    public function testExecuteEventPropertiesException()
+    {
+        $this->formDataAnalytics->shouldReceive('getPostData')->andReturn([
+            'event'         => $this->event
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php]: Keep only one space after key

File: Test/Unit/FormDataAnalyticsTest.php [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+        $this->formDataAnalytics->shouldReceive('getPostData')->andThrow($this->apiError);
+
+        $this->formDataAnalytics->execute();
+    }
+    public function testGetPostDataNotEmpty()
+    {
+        $this->formDataAnalytics->shouldReceive('getRequest')->andReturn($this->requestInterface);
+        $this->requestInterface->shouldReceive('getPostValue')->andReturn([
+            'event'         => $this->event,
+            'properties'    => $this->eventProperties
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php]: Keep only one space after property and align event accordingly

File: Test/Unit/FormDataAnalyticsTest.php [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+            ]);
+        
+        $this->assertSame([
+            'event'         => $this->event,
+            'properties'    => $this->eventProperties
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php]: Keep only one space after property and align event accordingly

File: Test/Unit/FormDataAnalyticsTest.php [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+        ], $this->formDataAnalytics->getPostData());
+    }
+    public function testGetPostDataEmpty()
+    {
+        $this->formDataAnalytics->shouldReceive('getRequest')->andReturn($this->requestInterface);
+        $this->requestInterface->shouldReceive('getPostValue')->andReturn([]);
+        
+        $this->assertSame('{}', $this->formDataAnalytics->getPostData());
+    }
+}
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php]: Add line after eof

File: Test/Unit/FormDataAnalyticsTest.php [hunk 7]
```diff
@@ -0,0 +1,136 @@
...
+            'field_name'        => 'title',
+            'field_type'        => 'text',
+            'platform'          => 'Magento',
+            'platform_version'  => '2.4.5-p1',
+            'plugin'            => 'Razorpay',
+            'plugin_version'    => '4.0.2',
+            'mode'              => 'test',
+            'ip_address'        => ''
+        ];
+    }
```
Comment (garimaprachi) [inline Test/Unit/FormDataAnalyticsTest.php:92]: Add spaces after each function end

**Review by garimaprachi — COMMENTED** [review]
LGTM

**Review by garimaprachi — COMMENTED** [review]
LGTM

**Review by abdulwahidsharief — APPROVED** [review]
LGTM

**Pull request merged**